Publisher/Sources/Services/DefaultAblyPublisherService.swift

```diff
@@ -85,6 +87,17 @@ class DefaultAblyPublisherService: AblyPublisherService {
             self.delegate?.publisherService(sender: self, didFailWithError: error.toErrorInformation())
         }
     }
+    
+    private func createARTMessage(for locationUpdate: EnhancedLocationUpdate, and batteryLevel: Float?) -> ARTMessage? {
+        do {
+            let geoJson = try EnhancedLocationUpdateMessage(locationUpdate: locationUpdate, batteryLevel: batteryLevel)
+            let data = try [geoJson].toJSONString()
+            return ARTMessage(name: EventName.enhanced.rawValue, data: data)
+        } catch let error {
+            self.delegate?.publisherService(sender: self, didFailWithError: ErrorInformation(error: error))
```

Comment:
This feels like a side effect. I'm not sure what pattern you're following but I'm not entirely comfortable with the setting, nor the method name. 🤔

Reply:
here https://github.com/ably/ably-asset-tracking-swift/pull/215